code/web/src/modules/profileForm/ProfileForm.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { Component } from 'react';
+import { connect } from 'react-redux';
+import Button from '../../ui/button';
+import { saveProfile } from '../user/api/actions';
+import { changeEditMode } from '../user/api/actions';
+import Availability from '../availability/Availability';
+
+
+// import { updateAvailability } from '../subscription/api/actions';
+
+class ProfileForm extends Component{
+  constructor(props) {
+    super(props)
+
+      this.state = {
+        img: '',
+        address: '',
+        email: '',
+        description: ''
+      }
+  }
+
+  updateProfileState(e) {
+    this.setState({ [e.target.name]: e.target.value })
```

Comment:
For backend teammates: `setState` allows us to update the data in the constructor's `this.state`.  So, line 25 says "Update the value (`e.target.value`) of the key (`e.target.name`) in state that matches the target (input)'s `name` attribute.  This is a way to dynamically update the state based on whichever input the user types into.